Solutions/moritz_w/Lesson_2/lesson2.py

```diff
@@ -0,0 +1,65 @@
+import sys
+
+def main():
+    test_list = [0, 1, 2]
+    append_list = [3, 4]
+
+    try:
+        print("new_list1 = " + str(test_list) + " + 3")
+        new_list1 = test_list + 3
+        print("Result: " + str(new_list1))
+    except TypeError as e:
+        print(e)
+
+    try:
+        print("new_list2 = " + str(test_list) + " + " + str(append_list))
+        new_list2 = test_list + append_list
+        print("Result: " + str(new_list2))
+    except TypeError as e:
+        print(e)
+
+    print("Change type of a")
+    print("List:")
+    a = []
+    print(type(a))
+    print("Change type of a to bool")
+    a = True
+    print(type(a))
+
+    print("--- Mutable Types ---")
+    print("Test list: " + str(test_list))
+    print("append new list")
+    test_list += append_list
```

Comment:
You can check that using dir() :). 

The method is called \_\_iadd\_\_().
Actually I've seen it only in list type, but there might other types that use it. 

I think when you use += python calls iadd if it is defined otherwise it calls add. So thanks to this you can implement custom behavior for += operator without changing + operator.